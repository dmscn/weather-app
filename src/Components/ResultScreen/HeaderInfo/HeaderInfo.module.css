.Container {
  text-align: center;
}

.Header {
  font-size: 2.5em;
  margin-bottom: .1em;
}

.Subtitle {
  font-size: 1em;
  color: #cecece;
}