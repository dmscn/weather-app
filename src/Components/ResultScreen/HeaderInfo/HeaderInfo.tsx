import React, { Component } from 'react'
import moment from 'moment';
import 'moment/min/locales';

import styles from './HeaderInfo.module.css';

moment.locale('pt-br');

export interface Props {
  location: string,
}

export default class HeaderInfo extends Component<Props> {
  render(): JSX.Element {
    const { location } = this.props;
    return (
      <section className={styles.Container}>
        <h1 className={styles.Header}>{location}</h1>
        <h4 className={styles.Subtitle}>{moment().format('LLLL')}</h4>
      </section>
    )
  }
}
