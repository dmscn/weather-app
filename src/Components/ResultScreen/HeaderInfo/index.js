export {default} from './HeaderInfo';
export * from './HeaderInfo';