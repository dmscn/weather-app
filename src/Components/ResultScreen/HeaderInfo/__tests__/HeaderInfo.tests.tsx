import React from 'react'
import {shallow} from 'enzyme';

import HeaderInfo from '../HeaderInfo';

describe('<HeaderInfo />', () => {
  it('renders', () => {
    const wrapper = shallow(<HeaderInfo location="Rio de Janeiro" />);
    expect(wrapper).toMatchSnapshot();
  });
});