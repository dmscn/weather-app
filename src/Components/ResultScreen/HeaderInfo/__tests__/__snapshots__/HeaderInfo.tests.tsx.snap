// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderInfo /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderInfo
    location="Rio de Janeiro"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="Header"
        >
          undefined, undefined
        </h1>,
        <h4
          className="Subtitle"
        >
          Segunda-feira, 18 de Fevereiro de 2019 às 19:16
        </h4>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "undefined, undefined",
          "className": "Header",
        },
        "ref": null,
        "rendered": "undefined, undefined",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Segunda-feira, 18 de Fevereiro de 2019 às 19:16",
          "className": "Subtitle",
        },
        "ref": null,
        "rendered": "Segunda-feira, 18 de Fevereiro de 2019 às 19:16",
        "type": "h4",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="Header"
          >
            undefined, undefined
          </h1>,
          <h4
            className="Subtitle"
          >
            Segunda-feira, 18 de Fevereiro de 2019 às 19:16
          </h4>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "undefined, undefined",
            "className": "Header",
          },
          "ref": null,
          "rendered": "undefined, undefined",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Segunda-feira, 18 de Fevereiro de 2019 às 19:16",
            "className": "Subtitle",
          },
          "ref": null,
          "rendered": "Segunda-feira, 18 de Fevereiro de 2019 às 19:16",
          "type": "h4",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
